File: 541-reverse-string-ii/reverse-string-ii.go
func reverseStr(s string, k int) string {
    str := []byte(s)
	n := len(str)

	for i := 0; i < n; i += 2 * k {
		
		end := i + k
		if end > n {
			end = n
		}
		reverse(str[i:end])
	}

	return string(str)
}

func reverse(b []byte) {
	l, r := 0, len(b)-1
	for l < r {
		b[l], b[r] = b[r], b[l]
		l++
		r--
	}
}

File: 541-reverse-string-ii/reverse-string-ii.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        res = list(s)

        for i in range(0, len(res), 2*k):
            res[i:i+k] = res[i:i+k][::-1]

        return "".join(res)